Lib/test/_test_multiprocessing.py

```diff
@@ -2682,6 +2682,9 @@ def test_make_pool(self):
                 p.join()
 
     def test_terminate(self):
+        if self.TYPE == 'threads':
+            self.skipTest("Threads cannot be terminated")
```

Comment:
Note the changes in this file are probably not necessary, it's just that without them the threads emulation of processes takes a _very_ long time to test. The reason is simple and not related to this PR: while you can terminate a process early, you cannot do that on a thread, so joining a sleeping thread has to wait for the sleep to finish.

Reply:
We can use a shorter wait time rather than skip the test entirely: https://github.com/python/cpython/pull/114186